### ISSUE TITLE
XmlMediaTypeCodec throws exception at application startup due to private member access violation on super class

### ISSUE THREAD

**Issue by michaelstievenart** [micronaut-projects/micronaut-jackson-xml#11]
### Steps to Reproduce
1. Created a Micronaut Application 
2. Add micronaut-jackson-xml (v 1.0.0)
3. Add micronaut-discovery-client 2.0.0 M3
4. Add basic eureka config

```
eureka:
  client:
    registration:
      enabled: true
    defaultZone: "${EUREKA_HOST:localhost}:${EUREKA_PORT:8761}"

```

### Expected Behaviour

The micronaut xml module should not try and access private objectMapper on the super class directly. Instead use the supplied getter. 
`this.getObjectMapper().copy()`

### Actual Behaviour
` this.objectMapper.copy();`
 throws an exception at startup as it has private access on super class.
```
@Override
    public JacksonMediaTypeCodec cloneWithFeatures(JacksonFeatures jacksonFeatures) {
        ObjectMapper objectMapper = this.objectMapper.copy(); // Private access on super class
        jacksonFeatures.getDeserializationFeatures().forEach(objectMapper::configure);
        jacksonFeatures.getSerializationFeatures().forEach(objectMapper::configure);

        return new XmlMediaTypeCodec(objectMapper, applicationConfiguration, codecConfiguration);
    }
```

```
[main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: class io.micronaut.xml.jackson.codec.XmlMediaTypeCodec tried to access private field io.micronaut.jackson.codec.JacksonMediaTypeCodec.objectMapper (io.micronaut.xml.jackson.codec.XmlMediaTypeCodec and io.micronaut.jackson.codec.JacksonMediaTypeCodec are in unnamed module of loader 'app')
java.lang.IllegalAccessError: class io.micronaut.xml.jackson.codec.XmlMediaTypeCodec tried to access private field io.micronaut.jackson.codec.JacksonMediaTypeCodec.objectMapper (io.micronaut.xml.jackson.codec.XmlMediaTypeCodec and io.micronaut.jackson.codec.JacksonMediaTypeCodec are in unnamed module of loader 'app')
	at io.micronaut.xml.jackson.codec.XmlMediaTypeCodec.cloneWithFeatures(XmlMediaTypeCodec.java:57)
	at io.micronaut.http.client.netty.RxNettyHttpClientRegistry.lambda$getClient$1(RxNettyHttpClientRegistry.java:293)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.micronaut.http.client.netty.RxNettyHttpClientRegistry.getClient(RxNettyHttpClientRegistry.java:190)
	at io.micronaut.http.client.netty.RxNettyHttpClientRegistry.getClient(RxNettyHttpClientRegistry.java:141)
	at io.micronaut.http.client.netty.RxNettyHttpClientRegistry.getClient(RxNettyHttpClientRegistry.java:74)
	at io.micronaut.http.client.interceptor.HttpClientIntroductionAdvice.intercept(HttpClientIntroductionAdvice.java:135)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:69)
	at io.micronaut.retry.intercept.DefaultRetryInterceptor.intercept(DefaultRetryInterceptor.java:138)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:69)
	at io.micronaut.retry.intercept.RecoveryInterceptor.intercept(RecoveryInterceptor.java:73)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:69)
	at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:106)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:69)
	at io.micronaut.discovery.eureka.client.v2.AbstractEurekaClient$Intercepted.register(Unknown Source)
	at io.micronaut.discovery.eureka.registration.EurekaAutoRegistration.register(EurekaAutoRegistration.java:195)
	at io.micronaut.discovery.registration.AutoRegistration.onApplicationEvent(AutoRegistration.java:58)
	at io.micronaut.discovery.registration.AutoRegistration.onApplicationEvent(AutoRegistration.java:38)
	at io.micronaut.context.DefaultBeanContext.publishEvent(DefaultBeanContext.java:1248)
	at io.micronaut.http.server.netty.NettyHttpServer.lambda$fireStartupEvents$4(NettyHttpServer.java:497)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.micronaut.http.server.netty.NettyHttpServer.fireStartupEvents(NettyHttpServer.java:495)
	at io.micronaut.http.server.netty.NettyHttpServer.start(NettyHttpServer.java:340)
	at io.micronaut.http.server.netty.NettyHttpServer.start(NettyHttpServer.java:109)
	at io.micronaut.runtime.Micronaut.lambda$start$2(Micronaut.java:70)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:68)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:294)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:280)
```

### Environment Information
- **Operating System**: Windows 10
- **Micronaut Version:** 2.0.0.M3
- **JDK Version:** JDK 12

**Comment by graemerocher** [comment]
Fixed in 1.1.0

**Comment by michaelstievenart** [comment]
Thanks, my mistake, didn't realise the 2.0.0.M3 parent was using version 1.0.0
